### PR TITLE
Add private docker registry

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

**Comment by jerith** [comment]
The registry is running, but it needs to have certs and be accessible on a consular domain.

**Comment by JayH5** [comment]
:+1:

File: puppet/manifests/default.pp [hunk 1]
```diff
@@ -2,8 +2,35 @@
...
+  package { 'openssl': }
+  ->
+  file { '/var/docker-certs':
+    ensure => directory,
+  }
+  ->
+  openssl::certificate::x509 { 'docker-registry':
+    country => 'NT',
+    organization => 'seed-stack',
+    commonname => 'docker-registry.service.dc1.consul',
```
Comment (JayH5) [inline puppet/manifests/default.pp]: Hopefully we can just make this `docker-registry.service.consul` and things should resolve. Consul should default to `dc1`.

**Comment by JayH5** [comment]
LGTM other than the `dc1` thing. We need to investigate a proper secrets/password storage thing at some point... Hashicorp has [Vault](https://vaultproject.io) :wink:  Perhaps something for the new year, though.

File: puppet/manifests/default.pp [hunk 2]
```diff
@@ -4,6 +4,29 @@
...
+
+  # NOTE: This cert wrangling is only good for a single machine. We need some
+  # other mechanism to get our certs to the right place in a multi-node setup.
+  package { 'openssl': }
+  ->
+  openssl::certificate::x509 { 'docker-registry':
+    country => 'NT',
+    organization => 'seed-stack',
+    commonname => 'docker-registry.service.consul',
+    base_dir => '/var/docker-certs',
```
Comment (JayH5) [inline puppet/manifests/default.pp:20]: This still need to be here?

Comment (jerith) [inline puppet/manifests/default.pp:20]: Yes, that creates the cert we're using. The thing that changed in the most recent commit was how we tell the docker daemon about it.

(Using the system CA store is more flexible, and also requires less mucking about with hostnames in config dirs.)

Comment (JayH5) [inline puppet/manifests/default.pp:20]: I meant that line in particular that references `/var/docker-certs`.

Comment (jerith) [inline puppet/manifests/default.pp:20]: Yes. Removing the thing that created that directory was accidental. I'll put it back.

**Comment by JayH5** [comment]
:+1:

**Comment by JayH5** [comment]
:+1: hopefully for the last time :stuck_out_tongue:

**Pull request merged**